irc/channel.go

```diff
@@ -458,22 +461,19 @@ func (channel *Channel) Join(client *Client, key string, isSajoin bool, rb *Resp
 	channel.Names(client, rb)
 
 	if givenMode != 0 {
-		rb.Add(nil, client.server.name, "MODE", chname, modestr, nick)
+		rb.Add(nil, client.server.name, "MODE", chname, modestr, details.nick)
 	}
 
-	channel.history.Add(history.Item{
-		Type:        history.Join,
-		Nick:        nickmask,
-		AccountName: accountName,
-		Msgid:       realname,
-	})
-
 	// TODO #259 can be implemented as Flush(false) (i.e., nonblocking) while holding joinPartMutex
 	rb.Flush(true)
 
 	replayLimit := channel.server.Config().History.AutoreplayOnJoin
 	if replayLimit > 0 {
-		items := channel.history.Latest(replayLimit)
+		// don't replay the client's own events
```

Comment:
I'm not intimately familiar with the History process, but mind explaining this in a little more detail? Generally with channel histories, they're replayed entirely with the client's own events inline. What sorta thing does this accomplish while replaying the events?

Reply:
Oh yeah. Specifically, I wanted to avoid replaying the client's own JOIN line a second time (I noticed that this produced a confusing experience in Kiwi, and figured it might in others).

I think for something like `/history` or the `CHATHISTORY` command, the client should definitely see their own events. They should also *probably* see their own events during this kind of autoreplay, so I think I'll change this. Options:

1. Do the event history query first, before adding the client's JOIN to the history list. (I didn't want to do this because I thought it would be nice to have a deterministic history order, by doing the history item add while holding `joinPartMutex`.)
1. Change the query so it only excludes the client's JOINs, but includes other events.

Reply:
Ahhhh I see. Alright, so this question has been covered and had some back-and-forth in the initial chathistory batch conversations. I'd recommend taking a look through the comments in these two issues: https://github.com/ircv3/ircv3-specifications/pull/156 https://github.com/ircv3/ircv3-specifications/issues/293